### ISSUE TITLE
Fuzzer reports incorrect results when spilling is enabled.

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#3309]
Looks like aggregation with spilling sometimes produced incorrect results. Here is an example of 'map_agg' with 2 grouping keys:

```
/Users/mbasmanova/cpp/velox-1/cmake-build-debug/velox/exec/tests/velox_aggregation_fuzzer_test --logtostderr=1 --duration_sec 600 --enable_window --seed 1511697966
I20221118 12:13:07.818006 18072045 AggregationFuzzer.cpp:511] ==============================> Started iteration 0 (seed: 1511697966)
I20221118 12:13:07.824481 18072045 AggregationFuzzer.cpp:603] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (g1:VARBINARY, ROW["g1"]), (p2:ARRAY<BOOLEAN>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, g1:VARBINARY, p2:ARRAY<BOOLEAN>
  -- Aggregation[SINGLE [g0, g1] a0 := map_agg(ROW["c0"],ROW["c1"]) mask: m0] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
    -- Values[1000 rows in 10 vectors] -> c0:BOOLEAN, c1:REAL, m0:BOOLEAN, g0:BOOLEAN, g1:VARBINARY
I20221118 12:13:07.836275 18072045 AggregationFuzzer.cpp:618] [ROW ROW<g0:BOOLEAN,g1:VARBINARY,p2:ARRAY<BOOLEAN>>: 296 elements, no nulls]

I20221118 12:13:07.866609 18072045 AggregationFuzzer.cpp:203] Testing plan #2 with spilling

I20221118 12:13:07.866641 18072045 AggregationFuzzer.cpp:603] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (g1:VARBINARY, ROW["g1"]), (p2:ARRAY<BOOLEAN>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, g1:VARBINARY, p2:ARRAY<BOOLEAN>
  -- Aggregation[FINAL [g0, g1] a0 := map_agg("a0")] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
    -- LocalPartition[REPARTITION] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
      -- Aggregation[PARTIAL [g0, g1] a0 := map_agg(ROW["c0"],ROW["c1"]) mask: m0] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
        -- Values[300 rows in 3 vectors] -> c0:BOOLEAN, c1:REAL, m0:BOOLEAN, g0:BOOLEAN, g1:VARBINARY
      -- Aggregation[PARTIAL [g0, g1] a0 := map_agg(ROW["c0"],ROW["c1"]) mask: m0] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
        -- Values[300 rows in 3 vectors] -> c0:BOOLEAN, c1:REAL, m0:BOOLEAN, g0:BOOLEAN, g1:VARBINARY
      -- Aggregation[PARTIAL [g0, g1] a0 := map_agg(ROW["c0"],ROW["c1"]) mask: m0] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
        -- Values[200 rows in 2 vectors] -> c0:BOOLEAN, c1:REAL, m0:BOOLEAN, g0:BOOLEAN, g1:VARBINARY
      -- Aggregation[PARTIAL [g0, g1] a0 := map_agg(ROW["c0"],ROW["c1"]) mask: m0] -> g0:BOOLEAN, g1:VARBINARY, a0:MAP<BOOLEAN,REAL>
        -- Values[200 rows in 2 vectors] -> c0:BOOLEAN, c1:REAL, m0:BOOLEAN, g0:BOOLEAN, g1:VARBINARY

I20221118 12:13:07.886608 18072045 AggregationFuzzer.cpp:618] [ROW ROW<g0:BOOLEAN,g1:VARBINARY,p2:ARRAY<BOOLEAN>>: 296 elements, no nulls]


Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Logically equivalent plans produced different results
Retriable: False
Expression: assertEqualResults({expected.result}, {actual.result})
Function: testPlan
File: /Users/mbasmanova/cpp/velox-1/velox/exec/tests/AggregationFuzzer.cpp
Line: 846


/Users/mbasmanova/cpp/velox-1/velox/exec/tests/utils/QueryAssertions.cpp:943: Failure
Value of: false
  Actual: false
Expected: true
Expected 296, got 296
2 extra rows, 2 missing rows
2 of extra rows:
      false | null | [false,false,true,true]
      true | null | [false,false,false,true,true,true]

2 of missing rows:
      false | null | [false,true]
      true | null | [false,true]
```

**Comment by mbasmanova** [comment]
Turns out this is due to MapVector::prepareForReuse() failing to reset sortedKeys_ to false.

**Comment by mbasmanova** [comment]
Fixed in #3291
